src/core/document/json.pl

```diff
@@ -2019,28 +2021,33 @@
         _).
 
 replace_document(DB, Document) :-
-    replace_document(DB, Document, _).
+    replace_document(DB, Document, false, _).
 
-replace_document(Transaction, Document, Id) :-
+replace_document(DB, Document, Id) :-
+    replace_document(DB, Document, false, Id).
+
+replace_document(Transaction, Document, Create, Id) :-
     is_transaction(Transaction),
     !,
     json_elaborate(Transaction, Document, Elaborated),
     get_dict('@id', Elaborated, Id),
```

Comment:
This requires the id to be present. If we want this to work like an insert for new objects, shouldn't we also have it generate the id when it is missing? Otherwise, the burden is on the client code or the user to generate the proper id that would otherwise have been generated during a normal insert call.

Reply:
This is after elaboration, so the id will exist at this point.